ts/Extensions/Annotations/Types/TimeCycles.ts

```diff
@@ -0,0 +1,305 @@
+/* *
+ *
+ *  Authors: Rafal Sebestjanski and Pawel Lysy
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type PositionObject from '../../../Core/Renderer/PositionObject';
+
+import Annotation from '../Annotations.js';
+import CrookedLine from './CrookedLine.js';
+import ControlPoint from '../ControlPoint.js';
+import U from '../../../Core/Utilities.js';
+import MockPointOptions from '../MockPointOptions';
+const { merge, isNumber, defined } = U;
+
+/**
+ * Internal types.
+ * @private
+ */
+declare global {
+    namespace Highcharts {
+        interface AnnotationControllable {
+            secondLineEdgePoints: [Function, Function];
+        }
+    }
+}
+interface TimeCyclesOptions extends CrookedLine.Options {
+    xAxis: number;
+    yAxis: number;
+}
+
+/**
+ * Function to create start of the path.
+ * @param {number} x x position of the TimeCycles
+ * @param {number} y y position of the TimeCycles
+ * @return {string} path
+ */
+function getStartingPath(x: number, y: number): string {
+    return `M ${x} ${y}`;
+}
+
+/**
+ * function to create circle paths
+ *
+ * @param {number} pixelInterval diameter of the circle in pixels
+ * @param {number} numberOfCircles number of cricles
+ * @return {string} path
+ */
+function getCirclePath(pixelInterval: number, numberOfCircles: number): string {
+    const strToRepeat = `a 1 1 0 1 1 ${pixelInterval} 0 `;
+    let path = strToRepeat;
+    for (let i = 1; i < numberOfCircles; i++) {
+        path += strToRepeat;
+    }
+
+    return path.trim();
+}
+
+/* *
+ *
+ *  Class
+ *
+ * */
+
+/* eslint-disable no-invalid-this, valid-jsdoc */
+
+class TimeCycles extends CrookedLine {
+
+    public init(annotation: Annotation, options: TimeCyclesOptions, index?: number): void {
+        if (defined(options.yAxis)) {
+            (options.points as MockPointOptions[]).forEach((point): void => {
+                point.yAxis = options.yAxis;
+            });
+        }
+
+        if (defined(options.xAxis)) {
+            (options.points as MockPointOptions[]).forEach((point): void => {
+                point.xAxis = options.xAxis;
+            });
+        }
+        super.init.call(this, annotation, options, index);
+
+    }
+
+    public getPath(): string {
+
+        return `${getStartingPath(this.startX, this.y)} ${getCirclePath(
+            this.pixelInterval,
+            this.numberOfCircles
+        )}`;
+    }
+
+    public addShapes(): void {
+        const typeOptions = this.options.typeOptions;
+        this.setPathProperties();
+        const shape = this.initShape(
+            merge(typeOptions.line, {
+                type: 'path',
+                d: this.getPath(),
+                points: this.options.points
+            }),
+            0
+        );
+
+        typeOptions.line = shape.options;
+    }
+
+    public addControlPoints(): void {
+
+        const options = this.options,

```

Comment:
Please decide and unify the spaces at the begging of the method/if.
I prefer no extra line here but it's up to you ;)

Reply:
Still visible